subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -537,4 +545,36 @@ apply from:'scriptPlugin.gradle'
         executedTasks == [':createFile']
         skippedTasks.empty  // The equals implementation for custom type always return false
     }
+
+    // Helper function that generate the FooType definition that can be used in the exact same way
+    private static String createFooTypeDefinitionAsEnum() {
+        return """
+            |enum FooType {
```

Comment:
Is it necessary to have the pipe character here? I don't think we should do that. It should just return the class definition without it.